libclamav/others.h

```diff
@@ -1174,6 +1174,19 @@ uint8_t cli_set_debug_flag(uint8_t debug_flag);
     } while (0)
 #endif
 
+#ifndef CLI_STRDUP
+#define CLI_STRDUP(buf, var, ...) \
+    do {                          \
+        var = cli_strdup(buf);    \
+        if (NULL == var) {        \
```

Comment:
This (and the below) macro jumps straight to `done` after an allocation fails, but doesn't report the error. So, the reason for the function not doing something (or not finishing) would be unknown.  I recognize that these macros are used in areas where we count on a certain structure (i.e., that a `done` label exists, and it's safe to jump straight there, but I wonder if it wouldn't be prudent to incorporate some kind of error reporting as well.